lua/telescope/finders.lua

```diff
@@ -248,6 +248,38 @@ function StaticFinder:new(opts)
   return setmetatable({ results = results }, self)
 end
 
+function StaticFinder:new_tree(opts)
+  assert(opts, "Options are required. See documentation for usage")
+
+  local input_results
+  if vim.tbl_islist(opts) then
+    input_results = opts
+  else
+    input_results = opts.results
+  end
+
+  local entry_maker = opts.entry_maker or make_entry.gen_from_tree()
+
+  assert(input_results)
+  assert(input_results, "Results are required for static finder")
+  assert(type(input_results) == 'table', "self.results must be a table")
+
+  local results = {}
+  local counter = 0
+  for k, v in pairs(input_results) do
+    local entry = entry_maker(k, v)
+
+    if entry then
+      -- entry.index = counter
+      table.insert(results, entry)
+    end
+
+    counter = counter + 1
+  end
+
+  return setmetatable({ results = results }, self)
+end
+
```

Comment:
I believe we don't need a new finder for this. Just do that in `internal.menu`

Reply:
I agree with not using another finder. I was not very familiar with the code base so I was trying out different ways to implement this. I think I have settled on using a different entry maker for the tree menu instead of creating a whole new finder.

Reply:
I have an issue though, how would I use the regular static finder instead creating a new one. My entry maker accepts two arguments while the entry maker in the regular static finder can only accept one argument. Also what would I set the index to because the regular static finder sets the index to the value which is not correct for the tree.

Reply:
I though you can do something like this before the picker. Correct me if i am wrong i am just rambling.
```lua
local level = opts.menu
if opts.passed_level then
  level = opts.menu[opts.passed_level] -- that needs some more work for levels below 1. Maybe passed_level should be a table like that. [ level1_key, level2_key, level3_key ] and we just table.insert newest level when a branch is selected and if branch == '..' then remove last level.
end
local results = {}
for k, v in pairs(opts.menu) do
  if type(v) == "string" then -- leaf
    table.insert(results, { leaf = v, menu = opts.menu }) -- and have each element have the table. So you also have it when we hit enter
  elseif type(k) == "string" then
    table.insert(results, { branch = k, menu = opts.menu }) -- when selecting you can then just check if branch is not nil. if thats the case the restart menu
  else
    -- we should never get here. Am i correct?
  end
end
table.insert(results, { branch = '..', menu = opts.menu })
```

And in entry maker you could do
```lua
function(entry)
  return {
    value = entry,
    display = display.leaf or display.branch,
    ordinal = display.leaf or display.branch,
  }
end
```
Does that makes sense to you or should i elaborate a little bit more. Also i don't get what you mean with index 😆

Reply:
Wow that is much better. I will start implementing that. Just so I am understanding we are going to preprocess the tree from something more user-friendly to an internal representation that we can use with an entry maker.

Reply:
Also the passed level thing I am implementing right now. We should use a stack like you said so it can remember multiple `..`

Reply:
Basically i was thinking we just have to reduce the table to the current level we are looking at. That was my idea. But my code might not work i just wrote that inside github. So let me know if you run into issues :)

I am also thinking we can do something cool and can display the rest of the tree of the current selected branch in the previewer. There is an interface so that every entry can have a defined function which then will be called and allows us to do `vim.api.nvim_buf_set_lines` for the previewer. We don't use that interface here anymore so there is no example. So feel free to ask me for information if you wanna do it.

Reply:
Don't worry your comments are not annoying :smile:. They are helping me a lot especially since I am new to all this stuff. Actually this is my first ever pull request so your guidance is really helpful.